src/main/java/io/vertx/core/CompositeFuture.java

```diff
@@ -72,8 +72,8 @@
    *
    * When the list is empty, the returned future will be already completed.
    */
-  static CompositeFuture all(List<Future> futures) {
-    return CompositeFutureImpl.all(futures.toArray(new Future[0]));
+  static CompositeFuture all(List<Future<?>> futures) {
```

Comment:
it has been rolled back because it is too breaking.

I think instead we should simply have the fixed methods on the `Future` interface directly and deprecate these.

Reply:
see my comment above

Reply:
okay thanks for your reply :)